components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/StreamingListener.java

```diff
@@ -0,0 +1,215 @@
+                builder.maxPingsOut(Integer.parseInt(maxPingsOut));
+            }
+
+            if (StringUtils.isNotEmpty(pingInterval)) {
+                builder.pingInterval(Duration.ofSeconds(Integer.parseInt(pingInterval)));
+            }
+
+            if (Boolean.parseBoolean(traceConnection)) {
+                builder.traceConnection();
+            }
+
+            StreamingConnectionFactory connectionFactory = new StreamingConnectionFactory(builder.build());
+            connection = connectionFactory.createConnection();
+        }
+        return true;
+    }
+
+    /**
+     * Consume the message received and inject into the sequence.
+     *
+     * @param sequenceName the sequence to inject the message to.
+     */
+    @Override public void consumeMessage(String sequenceName)
+            throws InterruptedException, IOException, TimeoutException {
+        SubscriptionOptions.Builder subscriptionOptions = new SubscriptionOptions.Builder();
+
+        String durableName = natsProperties.getProperty(NatsConstants.NATS_STREAMING_DURABLE_NAME);
+        String queueGroup = natsProperties.getProperty(NatsConstants.NATS_STREAMING_QUEUE_GROUP);
+        boolean isManualAck = Boolean.parseBoolean(natsProperties.getProperty(NatsConstants.NATS_STREAMING_MANUAL_ACK));
+        String ackWait = natsProperties.getProperty(NatsConstants.NATS_STREAMING_ACK_WAIT);
+        String maxInFlight = natsProperties.getProperty(NatsConstants.NATS_STREAMING_MAX_IN_FLIGHT);
+        String subscriptionTimeout = natsProperties.getProperty(NatsConstants.NATS_STREAMING_SUBSCRIPTION_TIMEOUT);
+        String dispatcher = natsProperties.getProperty(NatsConstants.NATS_STREAMING_DISPATCHER);
+
+        if (StringUtils.isNotEmpty(durableName)) {
+            subscriptionOptions.durableName(durableName);
+        }
+
+        if (isManualAck) {
+            subscriptionOptions.manualAcks(); // if subscriptionOptions.manualAcks() is not set, it will auto ack
+        }
+
+        if (StringUtils.isNotEmpty(ackWait)) {
+            subscriptionOptions.ackWait(Duration.ofSeconds(Integer.parseInt(ackWait)));
+        }
+
+        if (StringUtils.isNotEmpty(maxInFlight)) {
+            subscriptionOptions.maxInFlight(Integer.parseInt(maxInFlight));
+        }
+
+        if (StringUtils.isNotEmpty(subscriptionTimeout)) {
+            subscriptionOptions.subscriptionTimeout(Duration.ofSeconds(Integer.parseInt(subscriptionTimeout)));
+        }
+
+        if (StringUtils.isNotEmpty(dispatcher)) {
+            subscriptionOptions.dispatcher(dispatcher);
+        }
+
+        subscription = connection.subscribe(subject, queueGroup, natsMessage -> {
+            String message = new String(natsMessage.getData());
+            printDebugLog("Message Received to NATS Inbound EP: " + message);
+            log.info("Message Received to NATS Inbound EP: " + message);
+            boolean isInjected = injectHandler.invoke(message.getBytes(), sequenceName, null, null);
```

Comment:
We are passing a null connection here. would the NatsReplySender work in the expected way with a null connection?

Reply:
No there is no support for that. That is only with Core NATS. And Core NATS does not have acknowledgements and other features as Streaming.